client/src/components/ResourceHub/EditResourceModal.tsx

```diff
@@ -0,0 +1,145 @@
+import React, { ReactElement, FC, Dispatch, useEffect } from 'react';
+import { Modal, Button, Form, Input, Checkbox } from 'semantic-ui-react';
+import { IResource } from 'ssw-common';
+
+import { editResource, isError } from '../../utils/apiWrapper';
+
+import '../../css/EditResourceModal.css';
+
+const teams = [
+  'General',
+  'Editing',
+  'Factchecking',
+  'Illustration',
+  'Photography',
+  'Onboarding',
+  'Visuals',
+  'Writing',
+];
+
+interface IProps {
+  isOpen: boolean;
+  resource: IResource | null;
+  closeModal: Dispatch<void>;
+}
+
+const EditResourceModal: FC<IProps> = ({
+  isOpen,
+  resource,
+  closeModal,
+}): ReactElement => {
+  const [open, setOpen] = React.useState<boolean>(false);
+  const [resourceURL, setResourceURL] = React.useState<string>(
+    resource ? resource.link : '',
+  );
+  const [selectedTags, setSelectedTags] = React.useState<Set<string>>(
+    new Set(),
+  );
+
+  function handleTagSelect(tag: string): void {
+    const newTags = new Set(selectedTags);
+    if (newTags.has(tag)) {
+      newTags.delete(tag);
+    } else {
+      newTags.add(tag);
+    }
+    setSelectedTags(newTags);
+  }
+
+  async function modifyResource(): Promise<void> {
+    const editedResource = {
+      name: resource ? resource.name : null,
+      link: resourceURL,
+      teamRoles: Array.from(selectedTags),
+    };
+
+    const res = await editResource(resource?._id, editedResource);
+    if (!isError(res)) {
+      setOpen(false);
+    }
+  }
+
+  useEffect(() => {
+    if (resource) {
+      // Populate selectedTags set with resource's current teams
+      const newTags = new Set<string>();
+      for (const idx in Object.keys(resource.teamRoles)) {
+        newTags.add(resource.teamRoles[idx]);
+        setSelectedTags(newTags);
+      }
+    }
+  }, [resource]);
+
+  useEffect(() => {
+    setOpen(isOpen);
+  }, [isOpen]);
+
+  function close(): void {
+    setOpen(false);
+    closeModal();
+  }
+
+  function isTeam(team: string): boolean {
+    if (resource && selectedTags.has(team)) {
```

Comment:
Just `return resource && selectedTags.has(team)`. Even better:

`const isTeam = (team: string) => resource && selectedTags.has(team)`

Reply:
I don't think I can do this because resource could be undefined ?

Reply:
Danielle's right, this results in a typescript error